src/core/core-varlink.c

```diff
@@ -500,12 +501,17 @@ static void vl_disconnect(sd_varlink_server *s, sd_varlink *link, void *userdata
                 m->managed_oom_varlink = sd_varlink_unref(link);
 }
 
-static int manager_setup_varlink_server(Manager *m, sd_varlink_server **ret) {
+int manager_setup_varlink_server(Manager *m) {
         _cleanup_(sd_varlink_server_unrefp) sd_varlink_server *s = NULL;
         int r;
 
         assert(m);
-        assert(ret);
+
+        if (m->varlink_server)
+                return 0;
+
+        if (!MANAGER_IS_SYSTEM(m))
+                return -EPROTOTYPE;
```

Comment:
strange error code?

Reply:
I don't think it's weird? Or `ENOEXEC` could be another candidate.

Reply:
Well a user would see in the logs `Protocol wrong type for socket` which is not exactly helpful - just EINVAL would be clearer no?

Reply:
Quite a bunch of stuff would fail with -EINVAL... But I changed it anyways.